pkg/api/types.go

```diff
@@ -1380,3 +1381,15 @@ const (
 
 	PortHeader = "port"
 )
+
+// ResourceStats represents compute resource usage metrics associated with a pod or a container.
+type ResourceStats struct {
+	// Detailed instantaneous metrics.
+	Detailed *cadvisor_api.ContainerInfo `json:"detailed,omitempty"`
```

Comment:
cc/ @bgrant0607 we need to make the final decision on how to introduce resource related types to our API?

@vishh I am not comfortable to use cadvisor's API directly here, since it is introduced an extra dependency to cadvisor. But up to Brian to make the final call.